crates/biome_js_analyze/src/analyzers/nursery/no_useless_lone_block_statements.rs

```diff
@@ -0,0 +1,168 @@
+use biome_js_syntax::{
+    AnyJsStatement, JsBlockStatement, JsFileSource, JsLabeledStatement, JsStatementList,
+    JsVariableStatement, TextRange,
+};
+use biome_rowan::{AstNode, AstNodeList, BatchMutationExt};
+
+declare_rule! {
+    /// Disallow unnecessary nested block statements.
+    ///
+    /// > In JavaScript, prior to ES6, standalone code blocks delimited by curly braces do not create a new scope and have no use.
+    /// > In ES6, code blocks may create a new scope if a block-level binding (let and const), a class declaration or a function declaration (in strict mode) are present. A block is not considered redundant in these cases.
+    ///
+    /// Source: https://eslint.org/docs/latest/rules/no-lone-blocks
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// {}
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// if (foo) {
+    ///   bar();
+    ///   {
+    ///     baz();
+    ///   }
+    /// }
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// while (foo) {
+    ///   bar();
+    /// }
+    /// ```
+    ///
+    pub(crate) NoUselessLoneBlockStatements {
+        version: "next",
+        name: "noUselessLoneBlockStatements",
+        recommended: false,
+    }
+}
+
+impl Rule for NoUselessLoneBlockStatements {
+    type Query = Semantic<JsBlockStatement>;
+    type State = TextRange;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let block = ctx.query();
+        let is_module = ctx.source_type::<JsFileSource>().is_module();
+
+        if let Some(_labeled_statement) = JsLabeledStatement::cast(block.syntax().parent()?) {
+            return None;
+        }
+
+        if in_control_structure(block) {
+            return None;
+        }
+
+        if block.statements().is_empty() {
+            return Some(block.range());
+        }
+
+        if block
+            .statements()
+            .iter()
+            .any(|statement| statement_has_block_level_declaration(&statement, is_module))
+        {
+            return None;
+        }
+
+        Some(block.range())
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, range: &Self::State) -> Option<RuleDiagnostic> {
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                range,
+                markup! {
+                    "This block statement doesn't serve any purpose and can be safely removed."
+                },
+            )
+                .note(markup! {
+                "Standalone block statements without any block-level declarations are redundant in JavaScript and can be removed to simplify the code."
+            }),
+        )
+    }
+
+    fn action(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<JsRuleAction> {
+        let block = ctx.query();
+
+        if let Some(stmts_list) = block.parent::<JsStatementList>() {
```

Comment:
Because there is no else condition, you can simply use the `?` operator
```suggestion
        let Some(stmts_list) = block.parent::<JsStatementList>()?;
```

Reply:
Yes, if the parent cannot be casted to a statements list, then it returns `None`. The `?` operator is a shortcut to return `None` early.

Reply:
Cool, thanks!